src/lib/storage/table.cpp

```diff
@@ -79,6 +90,35 @@ void Table::_add_value_segment_to_chunk(std::shared_ptr<Chunk>& chunk, const std
   });
 }
 
-void Table::compress_chunk(ChunkID chunk_id) { throw std::runtime_error("Implement Table::compress_chunk"); }
+void Table::_add_dictionary_segment_to_chunk(std::shared_ptr<Chunk>& chunk, const std::string& type,
+                                             const std::shared_ptr<BaseSegment>& value_segment) {
```

Comment:
We'd suggest renaming this parameter to segment (to match the code below) because we don't know that the type is "ValueSegment"